doc/stdenv/platform-notes.chapter.md

```diff
@@ -140,81 +118,61 @@ Some common issues when packaging software for Darwin:
 
-  - libiconv
-  - libresolv
-  - libsbuf
+#### How to use libiconv on Darwin
 
-  Other common libraries are available in Darwin-specific versions with modifications from Apple. Note that these packages may be made the default on Darwin in the future.
+The libiconv package is included in the SDK by default along with libresolv and libsbuf. You do not need to do anything to use these packages. They are available automatically. If your derivation needs the `iconv` binary, add the `libiconv` package to your `nativeBuildInputs` (or `nativeCheckInputs` for tests).
 
-  - ICU (compatible with the top-level icu package, but it also provides `libicucore.B.dylib` with an ABI compatible with the Darwin system version)
-  - libpcap (compatible with the top-level libpcap, but it includes Darwin-specific extensions)
+#### Dealing with `darwin.apple_sdk.frameworks`
 
-- The legacy SDKs packages are still available in the `darwin` package set under their existing names, but all packages in these SDKs (frameworks, libraries, etc) are stub packages for evaluation compatibility.
+You may see references to `darwin.apple_sdk.frameworks`. This is the legacy SDK pattern, and it is being phased out. All packages in `darwin.apple_sdk`, `darwin.apple_sdk_11_0`, and `darwin.apple_sdk_12_3` are stubs that do nothing. If your derivation references them, you can delete them. The default SDK should be enough to build your package.
 
-  In most cases, a derivation can be updated by deleting all of its SDK inputs (frameworks, libraries, etc). If you had to override the SDK, see below for how to do that using the new SDK pattern. If your derivation depends on the layout of the old frameworks or other internal details, you have more work to do.
+Note: the new SDK pattern uses the name `apple-sdk` to better align with nixpkgs naming conventions. The legacy SDK pattern  uses `apple_sdk`. You always know you are using the old SDK pattern if the name is `apple_sdk`.
 
-  When a package depended on the location of frameworks, references to those framework packages can usually be replaced with `${apple-sdk.sdkroot}/System` or `$SDKROOT/System`. For example, if you substituted `${darwin.apple_sdk.frameworks.OpenGL}/Library/Frameworks/OpenGL.framework` in your derivation, you should replace it with `${apple-sdk.sdkroot}/System/Library/Frameworks/OpenGL.framework` or `$SDKROOT/System/Library/Frameworks`. The latter is preferred because it supports using the SDK that is resolved when multiple SDKs are propagated (see above).
+Some derivations may depend on the location of frameworks in those old packages. To update your derivation to find them in the new SDK, use `$SDKROOT` instead in `preConfigure`.  For example, if you substitute `${darwin.apple_sdk.frameworks.OpenGL}/Library/Frameworks/OpenGL.framework` in `postPatch`, replace it with `$SDKROOT/System/Library/Frameworks` in `preConfigure`.
```

Comment:
Even nicer is to just replace `-F` with `-iframeworkwithsysroot`, a la https://gitlab.freedesktop.org/xorg/lib/libapplewm/-/commit/be972ebc3a97292e7d2b2350eff55ae12df99a42. Or in many cases this could be removed entirely, I think.

Reply:
It’s not an exhaustive list. The goal isn’t to specify alternatives but to help people deal with their packages’ build systems. I did add a comment about possibly being able to remove some substitutions.